app/controllers/rails_sitemap/sitemap_controller.rb

```diff
@@ -32,5 +33,12 @@ def set_routes
         route[:path][0..-11]
       end
     end
+
+    def set_current_domain
+      uri = URI.parse(request.original_url)
+      pre_html = uri.html_safe? ? 'https://' : 'http://'
```

Comment:
you could also substract the request.original_fullpath to the original_url to get the domain with the corresponding http header